### PR TITLE
Bug 2090885: Fix treating missing clusters in policy compliance status as compliant

### PR DESCRIPTION
**Pull request by sudomakeinstall2**
Now checks specifically if the cluster is compliant with policy and don't consider missing clusters as compliant.

### CODE REVIEW

File: controllers/clustergroupupgrade_controller.go [hunk 1]
```diff
@@ -1421,45 +1421,25 @@ func (r *ClusterGroupUpgradeReconciler) getPolicyClusterStatus(policy *unstructu
...
+	if err != nil {
+		return nil, fmt.Errorf("cannot obtain all the details about the clusters in the CR: %s", err)
 	}
-
-	// Filter only the clusters present in the current upgrade.
-	if filterClustersForUpgrade {
-		var nonCompliantClustersInUpgrade []string
-		allClustersForUpgrade, err := r.getAllClustersForUpgrade(ctx, clusterGroupUpgrade)
+	for _, cluster := range allClustersForUpgrade {
+		compliance, err := r.getClusterComplianceWithPolicy(cluster, policy)
```
Comment (jc-rh) [inline controllers/clustergroupupgrade_controller.go:1433]: we can't redo this for every single cluster. it's the same api call.

Comment (sudomakeinstall2) [reply controllers/clustergroupupgrade_controller.go:1433]: That makes sense however I kinda repeated what `getClustersNonCompliantWithManagedPolicies` does :laughing:

Comment (sudomakeinstall2) [reply controllers/clustergroupupgrade_controller.go:1433]: I am on it as soon as it passes the tests

Comment (jc-rh) [reply controllers/clustergroupupgrade_controller.go:1433]: Don't worry about it in this PR then. Let's get this fix in first.

**Comment by jc-rh** [comment]
It seems to be a conscious decision to treat missing cluster as compliant. We need to revisit the reasons behind that decision before changing this behavior.

**Review by jc-rh — APPROVED** [review]

**Pull request merged**